<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/4.1.1/animate.min.css">
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bulma/0.9.0/css/bulma.css">

  <title>About</title>
  <style media="screen">
    .hero{
      background: url('images/bird-background.jpg') center / cover
    }

    .pic-holder{
      overflow:auto
    }

    .in-pic{
      float: left;
      margin-right: 10px;
      margin-top: 5px;
    }

    #mut_1{
      color:orange;
    }
    #mut_2{
      color:green;
    }
  </style>
  <title>Results</title>
</head>
<body>
  <section class="hero is-fullheight">
    <div class="hero-body">
      <div class="block">

      </div>

      <div class="container">

        <div class="block">
            <a href="result.html" class="button is-dark is-medium animate__animated animate__flipInX animate__infinite">Results</a>
        </div>

          <div class="column">
            <p class="notification is-dark pic-holder">
              <img src="images/OR_CATAAT.png" alt="or_CAT" width="300" class="in-pic">
              In the OR-type, we set the rules for an OR system. We mutated two motifs this time:
              the TATAAA and a CATAAT motif. The TATAAA was mutated into a T<b id="mut_2">G</b>TAAA and the CATAAT was
              mutated into CA<b id="mut_1">C</b>AAT. Sequences can be expressed if they contain any of the mutations or
              both of the mutations. In the first image, a gene had the mutation on the CATAAT
              element. So we see the negative orange signal.<br>

              <img src="images/OR_TATAAA.png" alt="or_CAT" width="300" class="in-pic">
              In the second image the gene tested had the mutation on the TATAAA element hence the green signal shown.

              <br>

              <img src="images/OR_TAT_CAT.png" alt="or_CAT" width="300" class="in-pic">
              Finally a gene with the mutations on both elements, will have a mixture of green and orange signals
              which are distant from each other. At this point you may be wondering why we have two orange signals
              and just one green signal. All these signals are produced from the same gene, however these are different
              variants of the same gene because they come from different genomes. This means there is possibility of
              indels changing elements or shifting them within a sequence.
            </p>
          </div>

          <div class="column">
            <p class="notification is-dark pic-holder">
              <span class="tag is-link">Motif discovery</span>: We are also able to capture
              some important motifs using our CNNs. We could actually capture the GATA-disc promoter
              motif within the CBL interacting protein kinase 9, a gene involved in biosynthesis in
              <i>A. thaliana</i>
              <img src="images/AT1G01140.png" alt="AT1G01140" class="in-pic">
              CNNs are very powerful because of their repetitive use of filters. Since these filters can be
              predefined as the length of short motifs, we simply create filters which are 6 bp long and use
              these to scan an entire sequence. As the filters go through each sequence, specific
              mathematical calculations called convolutions will be performed. This produces feature maps at the
              end of each convolutional layer. One can think of a convolution as inputs multiplied by the weights
              plus biases. The weights are represented by our short windows which are 6 bp long.
              Once we train our network and do predictions we can feed these predictions to DeepLIFT which will
              help us visualize the regions which were crucial to prediction. The GATA-disc for example is
              represented by the pattern "TGATAA". This is captured within our predictions.
            </p>
          </div>


            <div class="column">
              <p class="notification is-warning">
                <span class="tag is-danger">Hint</span>
                The beautiful images and interpretation of our deep convolutional network we have is all thanks
                to the friendly folks who created the feature importance extraction tool
                 called <a href="https://arxiv.org/abs/1704.02685" target="_blank" class="has-text-info">Deeplift</a>.
                Deeplift computes importance scores by comparing each original sequence to a refernce input sequence, in
                our case a shuffled form of the original sequence.
              </p>
            </div>



        <footer class="card-footer">
          <p class="card-footer-item has-text-white animate__animated animate__pulse animate__infinite">&copy Network analysis and modelling (NAM)</p>
        </footer>

        <div class="block">
          <nav class="pagination">
            <a href="about.html" class="pagination-previous has-text-white animate__animated animate__pulse animate__infinite">previous</a>
            <a href="perspective.html" class="pagination-next has-text-white animate__animated animate__pulse animate__infinite">next</a>
            <ul class="pagination-list">
              <li><a href="result.html" class="pagination-link has-text-white is-current">3</a></li>
              <li><a href="perspective.html" class="pagination-link has-text-white">4</a></li>
            </ul>
          </nav>
        </div>

      </div>

    </div>

  </section>

</body>
</html>
